plugin/interestingwords.vim

```diff
@@ -197,29 +197,27 @@ function! s:buildColors()
   if (s:hasBuiltColors)
     return
   endif
-  if has('gui_running')
-    let ui = 'gui'
-    let wordColors = g:interestingWordsGUIColors
-  else
-    let ui = 'cterm'
-    let wordColors = g:interestingWordsTermColors
-  endif
+  let nGUIColors = len(g:interestingWordsGUIColors)
+  let nTermColors = len(g:interestingWordsTermColors)
+  let colorIndices = range(max([nGUIColors, nTermColors]))
```

Comment:
Sorry for the late response, you are absolutely right in everything you said. Hopefully the interface for the term gui colors doesn't change often enough in the future for the maintenance to become a bother :smile: .
Thank you for your input.